cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -205,6 +205,8 @@ public MasterServiceMain() {
     login(cConf);
 
     Configuration hConf = HBaseConfiguration.create();
+    hConf.set(Constants.HBase.CLIENT_RETRIES, cConf.get(Constants.HBase.MASTER_CLIENT_RETRIES));
```

Comment:
Shouldn't do it here. Do it in the `ConfigModule`. The right way to do it is.

1. In `cdap-default.xml`, have two properties, `hbase.client.retries.number` and `hbase.rpc.timeout`. Set them to `2` and `15000` respectively. These are the defaults used by the CDAP system.
2. In `ConfigModule.java`, copy the above two properties from `cConf` to `hConf` if it exists. Or we can just do what copying of tephra/twill properties does by copying all configs prefixed with `hbase.` from `cConf` to `hConf`.
3. In `AbstractDistributedProgramRunner`, in the constructor, make a copy for the `hConf` and `cConf`.
4. In the `run` method, unset the `hbase.client.retries.number` and `hbase.rpc.timeout` in `hConf` as well as in `cConf` before saving them to files.

Since those properties are not set in both `hConf` and `cConf`, the default from `hbase-site`/`hbase-default` will be used in the `TwillRunnable`.

I also notice we are set two tephra constants in the `AbstractProgramTwillRunnable`, which we shouldn't as well. Those should be done in `AbstractDistributedProgramRunner` as well, by setting those two properties in the copied `cConf` before localizing (in the runnable, it will be copied to `hConf` inside the `ConfigModule`, hence no need to unset the one in `hConf` in the program runner side).